src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -95,6 +95,43 @@ public static void VerifyAndUpdateFrameworkTypes(this IEnumerable<Schema> allSch
             }
         }
 
+        internal static string GetTypePropertyName(string parentName, string propertyTypeName)
```

Comment:
nit: can we call it `enclosingTypeName` since parent implies some sort of inheritance hierarchy with the property?